junit/kubernetes-server-mock/src/main/java/io/fabric8/kubernetes/client/server/mock/CustomResourceDefinitionProcessor.java

```diff
@@ -49,19 +58,80 @@ public void process(String path, String crdString, boolean delete) {
       return;
     }
     if (delete) {
-      extractor.removeCrdContext(context);
+      removeCrdContext(context);
     } else {
-      extractor.addCrdContext(context);
+      addCrdContext(context);
     }
   }
 
   public boolean isStatusSubresourceEnabledForResource(Map<String, String> pathValues) {
-    CustomResourceDefinitionContext context = extractor.getCrdContext(pathValues.get(KubernetesAttributesExtractor.API),
+    Optional<CustomResourceDefinitionContext> context = getCrdContext(pathValues.get(KubernetesAttributesExtractor.API),
         pathValues.get(KubernetesAttributesExtractor.VERSION), pathValues.get(KubernetesAttributesExtractor.PLURAL));
-    if (context == null) {
-      return false;
+    return context.map(CustomResourceDefinitionContext::isStatusSubresource).orElse(false);
+  }
+
+  private static List<String> key(CustomResourceDefinitionContext c) {
+    return key(c.getGroup(), c.getVersion());
+  }
+
+  private static List<String> key(String group, String version) {
+    return Arrays.asList(group, version);
+  }
+
+  public Optional<CustomResourceDefinitionContext> getCrdContext(String api, String version, String plural) {
+    return Optional.ofNullable(this.crdContexts.get(Arrays.asList(api, version))).map(v -> v.get(plural));
+  }
+
+  public Optional<CustomResourceDefinitionContext> findCrd(String api, String version, String kind) {
+    return Optional.ofNullable(crdContexts.get(Arrays.asList(api, version)))
+        .flatMap(m -> m.values().stream().filter(c -> c.getKind().equals(kind)).findFirst());
+  }
+
+  /**
+   * Remove the given context and return a collection of the remaining contexts for the api/version
+   */
+  public Optional<Collection<CustomResourceDefinitionContext>> removeCrdContext(CustomResourceDefinitionContext context) {
+    return Optional.ofNullable(this.crdContexts.computeIfPresent(key(context), (k, v) -> {
+      v.remove(context.getPlural());
+      return v;
+    })).map(Map::values);
+  }
+
+  /**
+   * Remove the given context and return a collection of the remaining contexts for the api/version
+   */
+  public Optional<Collection<CustomResourceDefinitionContext>> addCrdContext(CustomResourceDefinitionContext context) {
```

Comment:
nit, maybe javadoc doesn't match with what method is doing?

Reply:
Good catch.  It's now been simplified in general.